src/scenic/syntax/veneer.py

```diff
@@ -1612,10 +1613,29 @@ def VisibleFrom(base):
     if not isA(base, Point):
         raise TypeError('specifier "visible from O" with O not a Point')
 
+    def helper(self):
+        if mode2D:
+            position = Region.uniformPointIn(base.visibleRegion)
+        else:
+            # We can limit the potential positions to an overapproximation of the
+            # visible region.
+            hw = self.width / 2
+            hl = self.length / 2
+            hh = self.height / 2
+            radius = hypot(hw, hl, hh)
+
+            buffered_vr = base.visibleRegion._bufferOverapproximate(
+                radius / 2, BUFFERING_PITCH
+            )
+
+            position = Region.uniformPointIn(buffered_vr)
+
+        return {"position": position, "_observingEntity": base}
+
     return Specifier(
         "Visible/VisibleFrom",
         {"position": 3, "_observingEntity": 1},
-        {"position": Region.uniformPointIn(base.visibleRegion), "_observingEntity": base},
+        DelayedArgument({"width", "length", "height"}, helper),
```

Comment:
This should not depend on `width`, etc. in 2D mode. (Unlikely to cause problems, but no reason to break backwards compatibility here.) I'd check for 2D mode in the specifier rather than inside the delayed argument.

Reply:
Resolved by new approach where we move this logic to pruning.